backend/inspirehep/pidstore/providers/cnum.py

```diff
@@ -53,12 +55,41 @@ def next(cls, data):
 
         adjusted_date = datetime.strptime(opening_date, "%Y-%m-%d").strftime("%y-%m-%d")
         cnum = f"C{adjusted_date}"
-        cnums_count = PersistentIdentifier.query\
-            .filter(PersistentIdentifier.pid_value.like(f"{cnum}%"))\
-            .filter(PersistentIdentifier.pid_type == "cnum")\
-            .count()
 
-        if cnums_count > 0:
-            cnum = f"{cnum}.{cnums_count}"
+        retry_count = 0
+        try:
+            # Try to get postfix number from latest pid
+            last_cnum = (
+                PersistentIdentifier.query.filter(
+                    PersistentIdentifier.pid_value.like(f"{cnum}%")
+                )
+                .filter(PersistentIdentifier.pid_type == "cnum")
+                .order_by(PersistentIdentifier.created.desc())
```

Comment:
I think you're overcomplicating this. `created` is not very correlated with CNUM value ordering due to the migration inserting things in random order. What you want is to order by CNUM value, which happens to match lexicographic ordering on `pid_value`. so why don't you simply
```suggestion
                .order_by(PersistentIdentifier.pid_value.desc())
```

Reply:
because pid_value of "C20-01-01.20" would be smaller than "C20-01-01.3"